test/release/examples/primers/modules.chpl

```diff
@@ -0,0 +1,386 @@
+   a file is not enclosed in an explicit module, defined using the 'module'
+   keyword, then the file itself is treated as a module with the same name as
+   the file (minus the .chpl suffix).  The compiler can be directed to include
+   modules in separate files by naming them on the command line, utilizing the
+   -M flag, etc. (see the man page for exact details).
+ */
+module modToUse {
+
+  /* In this case, foo is a public module-level variable that is defined within
+     the module modToUse */
+  var foo = 12;
+
+  /* As is bar. */
+  var bar: int = 2;
+
+  /* A symbol can be declared "private" - this means that only code defined
+     within the same scope as the definition of this symbol can access it.
+
+     Here, hiddenFoo is a private global variable, which is only accessible by
+     symbols defined within modToUse
+  */
+  private var hiddenFoo = false;
+
+
+  /* baz is a public function which is defined within modToUse */
+  proc baz (x, y) {
+    return x * (x + y);
+  }
+
+  /* hiddenBaz is a private function, which is also only accessible by symbols
+     defined within modToUse.
+  */
+  private proc hiddenBaz(a) {
+    writeln(a);
+    return a + 3;
+  }
+
+  // In the current implementation, private cannot be applied to type
+  // definitions; type aliases, and declarations of enums, records, and
+  // classes cannot be declared private.  Private also cannot be applied to
+  // fields or methods yet.
+
+}
+
+module AnotherModule {
+  var a = false;
+}
+
+module ThirdModule {
+  var b = -13.0;
+}
+
+
+module Conflict {
+  /* This variable shares a name with a symbol in modToUse. */
+  var bar = 5;
+
+  var other = 5.0 + 3i;
+
+  var another = false;
+}
+
+
+module DifferentArguments {
+  /* This function shares a name with a function in modToUse, but requires
+     different arguments */
+  proc baz(x) {
+    return x - 2;
+  }
+}
+
+module MainModule {
+  proc main() {
+    writeln("Access from outside a module");
+
+    /* If a module is not the main module for a program, it is desirable for its
+       contents to be accessible to external modules.  There are several
+       strategies for accomplishing this:
+
+       First, a symbol can be referenced explicitly - this is done using the
+       module name and a separating '.' as a prefix to the name of the symbol
+       desired.
+    */
+    var thriceFoo = 3 * modToUse.foo; // should be 36
+    writeln(thriceFoo);
+
+
+    {
+      /* If many of the module's symbols are desired, or the same symbol is
+         desired multiple times, then it is best to utilize what is known as a
+         "use statement".
+      */
+      use modToUse;
+
+      /* Use statements can be inserted at any lexical scope that contains
+         executable code.
+
+         A use statement makes all of the module's visible symbols available to
+         the scope which contains the use statement.  These symbols may then be
```

Comment:
I'm not at all expert in them, but have generally been trained to prefer 'that' over 'which', and to think of using 'which' for cases when the clause could be dropped without breaking the completeness of the sentence.  I also use it to avoid duplicate 'that's in a single sentence, or when the clause feels like it ought to be set off by a comma (which maybe is the same criteria as "when the clause could be dropped"...)